### PR TITLE
TE7-T1151: Add Home Screen

### PR DESCRIPTION
**Pull request by DevarshBhalara**
- Create home feed page

### CODE REVIEW

File: My Spotify/My Spotify/Modules/Home/Controllers/HomeVC.swift [hunk 1]
```diff
@@ -9,10 +9,90 @@ import UIKit
...
+        
+        viewModel.getPlaylist()
+        viewModel.getAlbumbs()
+        viewModel.getFeaturedPlaylist()
+        tblHome.register(UINib(nibName: "TVHomeHeader", bundle: nil), forHeaderFooterViewReuseIdentifier: "TVHomeHeader")
+    }
+    
+    
+    private func setUpBind() {
+//        viewModel.success.bind { [weak self] playlist in
```
Comment (Kruna1Pate1) [inline My Spotify/My Spotify/Modules/Home/Controllers/HomeVC.swift]: Remove commented code

File: My Spotify/My Spotify/Modules/Home/Controllers/HomeVC.swift [hunk 2]
```diff
@@ -9,10 +9,90 @@ import UIKit
...
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        tableView.bounds.height / 3
+    }
+    
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        guard let sectionHeader = tableView.dequeueReusableHeaderFooterView(withIdentifier: "TVHomeHeader") as? TVHomeHeader else {
+            return UIView()
+        }
+        print("\(homeItems[section])")
```
Comment (Kruna1Pate1) [inline My Spotify/My Spotify/Modules/Home/Controllers/HomeVC.swift]: remove print
```suggestion
```

File: My Spotify/My Spotify/Modules/Home/Model/HomePlaylist.swift [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+struct HomePlaylist: Codable {
+    let href: String?
+    let items: [Item]?
+}
+
+// MARK: - Item
+struct Item: Codable {
+    let collaborative: Bool?
+    let description: String?
+//    let externalUrls: ExternalUrls?
```
Comment (Kruna1Pate1) [inline My Spotify/My Spotify/Modules/Home/Model/HomePlaylist.swift]: remove commented code

File: My Spotify/My Spotify/Modules/Home/Model/HomePlaylist.swift [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    let owner: Owner?
+    let primaryColor: String? = nil
+    let itemPublic: Bool?
+    let snapshotID: String?
+    let tracks: Tracks?
+    let type, uri: String?
+
+    enum CodingKeys: String, CodingKey {
+        case collaborative, description
+//        case externalUrls = "external_urls"
```
Comment (Kruna1Pate1) [inline My Spotify/My Spotify/Modules/Home/Model/HomePlaylist.swift]: Remove commented code

File: My Spotify/My Spotify/Modules/Home/ViewModel/HomeViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+//
+
+import Foundation
+class HomeViewModel {
+    
+    // MARK: - Variables
+    var success = Dynamic<[Item]?>(nil)
+    var fail = Dynamic<String>("")
+    var getAllData = Dynamic<[HomeDisplay]?>(nil)
+    var _getAllData = [HomeDisplay]()
```
Comment (Kruna1Pate1) [inline My Spotify/My Spotify/Modules/Home/ViewModel/HomeViewModel.swift]: no need to use backing field

File: My Spotify/My Spotify/Modules/Home/Views/CVCell/CVPlaylist.swift [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+//
+//  Created by Devarsh Bhalara on 05/07/23.
+//
+
+import UIKit
+
+class CVPlaylist: UICollectionViewCell {
+    
+    //MARK: Outlets
+    @IBOutlet weak var lblPlaylistName: UILabel!
```
Comment (Kruna1Pate1) [inline My Spotify/My Spotify/Modules/Home/Views/CVCell/CVPlaylist.swift]: Make outlets private

File: My Spotify/My Spotify/Modules/Home/Views/TVCell/TVHomeCell.swift [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+//
+//  Created by Devarsh Bhalara on 05/07/23.
+//
+
+import UIKit
+
+class TVHomeCell: UITableViewCell {
+    
+    // MARK: - Outlets
+    @IBOutlet weak var clvPlaylist: UICollectionView!
```
Comment (Kruna1Pate1) [inline My Spotify/My Spotify/Modules/Home/Views/TVCell/TVHomeCell.swift:13]: Make outlets private

File: My Spotify/My Spotify/Modules/Home/Views/TvHomeHeader/TVHomeHeader.swift [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+//
+//  Created by Devarsh Bhalara on 05/07/23.
+//
+
+import UIKit
+
+class TVHomeHeader: UITableViewHeaderFooterView {
+
+    // MARK: - Outlets
+    @IBOutlet weak var lblHeaderTitle: UILabel!
```
Comment (Kruna1Pate1) [inline My Spotify/My Spotify/Modules/Home/Views/TvHomeHeader/TVHomeHeader.swift]: Make Outlets private

File: My Spotify/My Spotify/Network Layer/APIManager+Request.swift [hunk 9]
```diff
@@ -11,6 +11,10 @@ import Alamofire
 enum RequestItemsType: Equatable {
     case authToken(code: String)
     case refreshToken
+
```
Comment (Kruna1Pate1) [inline My Spotify/My Spotify/Network Layer/APIManager+Request.swift]: Remove line break

File: My Spotify/My Spotify/Network Layer/APIManager+Request.swift [hunk 10]
```diff
@@ -19,38 +23,70 @@ enum RequestItemsType: Equatable {
...
+            return .get
+        case .authToken, .refreshToken:
+            return .post
+        }
     }
     
     var headers: HTTPHeaders? {
-        return []
+        switch self {
+        case .authToken, .refreshToken:
```
Comment (Kruna1Pate1) [inline My Spotify/My Spotify/Network Layer/APIManager+Request.swift]: Token are added through `adapt` in Interceptor.

File: My Spotify/My Spotify/Network Layer/APIManager.swift [hunk 11]
```diff
@@ -25,7 +25,7 @@ class APIManager {
             handler(.failure(CustomError(title: APIError.errorAlertTitle, body: APIError.noInternet)))
             return
         }
-        
+        print(type.url)
```
Comment (Kruna1Pate1) [inline My Spotify/My Spotify/Network Layer/APIManager.swift]: Remove

**Review by Kruna1Pate1 — CHANGES_REQUESTED** [review]

File: My Spotify/My Spotify/Modules/Home/Controllers/HomeVC.swift (hunk 2 above)
Comment (DevarshBhalara) [reply My Spotify/My Spotify/Modules/Home/Controllers/HomeVC.swift]: Done

**Review by Kruna1Pate1 — APPROVED** [review]

**Pull request merged**